datahub-web-react/src/app/entity/shared/tabs/Dataset/AccessManagement/utils.tsx

```diff
@@ -0,0 +1,27 @@
+export function handleAccessRoles(externalRoles, loggedInUser) {
+    const accessRoles = new Array<any>();
+    if (
+        externalRoles?.dataset?.access &&
+        externalRoles?.dataset?.access.roles &&
+        externalRoles?.dataset?.access.roles.length > 0
+    ) {
+        externalRoles?.dataset?.access?.roles?.forEach((userRoles) => {
+            const role = {
+                name: userRoles?.role?.properties?.name || ' ',
+                description: userRoles?.role?.properties?.description || ' ',
+                accessType: userRoles?.role?.properties?.type || ' ',
+                hasAccess:
+                    (userRoles?.role?.actors?.users &&
+                        userRoles?.role?.actors?.users.length > 0 &&
+                        userRoles?.role?.actors?.users?.some(
+                            (user) => user.user.urn === loggedInUser?.me?.corpUser.urn,
+                        )) ||
+                    false,
+                url: userRoles?.role?.properties?.requestUrl || window.location.href,
```

Comment:
minor: so if there's no requestUrl on the role we just send them to this page? I feel like the button should either not exist to request access or it should be disabled with a tooltip saying this role has no request access URL associated with it